vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/steps/ActionSteps.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2019-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.bdd.mobileapp.steps;
+
+import java.time.Duration;
+
+import org.jbehave.core.annotations.When;
+import org.vividus.bdd.steps.ui.validation.IBaseValidations;
+import org.vividus.mobileapp.action.TapActions;
+import org.vividus.ui.action.search.Locator;
+
+public class ActionSteps
+{
+    private final TapActions tapActions;
+    private final IBaseValidations baseValidations;
+
+    public ActionSteps(TapActions tapActions, IBaseValidations baseValidations)
+    {
+        this.tapActions = tapActions;
+        this.baseValidations = baseValidations;
+    }
+
+    /**
+     * Taps on <b>element</b> located by <b>locator</b> with specified <b>duration</b>
+     * The tap involves the following actions: press => wait => release
+     * @param locator locator to find an element
+     * @param duration between an element is pressed and released in
+     * <a href="https://en.wikipedia.org/wiki/ISO_8601">ISO_8601</a> format
+     */
+    @When("I tap on element located `$locator` with duration `$duration`")
+    public void tapByLocatorWithDuration(Locator locator, Duration duration)
+    {
+        baseValidations.assertElementExists("The element to tap", locator).ifPresent(e -> tapActions.tap(e, duration));
+    }
+
+    /**
+     * Taps on <b>element</b> located by <b>locator</b>
+     * The tap involves the following actions: press => release
+     * @param locator locator to find an element
+     */
+    @When("I tap on element located `$locator`")
+    public void tapByLocator(Locator locator)
+    {
+        tapByLocatorWithDuration(locator, Duration.ZERO);
```

Comment:
I would prefer not to create additional methods here

Reply:
if some users want to tap with zero duration (I have no idea why, but taking into account appium it could be anything), then we shouldn't modify the action